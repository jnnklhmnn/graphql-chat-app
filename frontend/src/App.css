.App {
  max-width: 300px;
  max-height: 504px;
  margin: 0 auto;
  margin-top: 200px;
  border: 1px solid #8a7f7f;
  border-radius: 5px;
  position: relative;
}

.message-section {
  height: 400px;
  overflow-y: scroll;
  background-color: #d4d0bc;
}

.App.initial-load-enter {
  transform: scale(0);
  will-change: transform;
}

.App.initial-load-enter-active {
  transform: scale(1);
  transition: transform 1s ease-in;
}