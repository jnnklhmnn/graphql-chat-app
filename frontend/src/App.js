import React, { Component } from "react";
import MessageInput from "./message-input.js";
import Counter from "./counter.js";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import Message from "./message.js";
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import "./App.css";

export class App extends Component {
  /**
   * Creates the Markup for the initial App including components,
   * based on the finished Server Request, returns empty div if request is pending
   *
   * @returns Markup
   */


  getApp = () => {
    if (this.props.allPostsQuery.loading) {
      return <div  key={'notloaded'}></div>;
    } else {
      return (
        <div className="App" key={'loaded'}>
          <Counter
            numberOfMessages={this.props.allPostsQuery.allPosts.length}
          />
          <div className="message-section">
            {this.props.allPostsQuery.allPosts.map(allPosts => (
              <React.Fragment key={allPosts.id}>
                <Message
                  message={allPosts.message}
                  updatedAt={allPosts.updatedAt}
                  createdAt={allPosts.createdAt}
                  id={allPosts.id}
                />
              </React.Fragment>
            ))}
          </div>
          <MessageInput />
        </div>
      );
    }
  };

  render() {
    const App = this.getApp();
    return <ReactCSSTransitionGroup
    transitionName="initial-load"
    transitionEnter={true}
    transitionLeave={false}
    transitionEnterTimeout={1000}>
    {App}
  </ReactCSSTransitionGroup>;
  }
}

export const ALL_POSTS_QUERY = gql`
  query {
    allPosts {
      id
      message
      createdAt
      updatedAt
    }
  }
`;

const AppWithQuery = graphql(ALL_POSTS_QUERY, {
  name: "allPostsQuery"
})(App);

export default AppWithQuery;
