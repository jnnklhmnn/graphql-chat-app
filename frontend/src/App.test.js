import { App } from "./App.js";
import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

describe("App", () => {
  
  test("renders in Loading State Correctly", () => {
    Enzyme.configure({ adapter: new Adapter() });
    const wrapper = shallow(<App allPostsQuery={{ loading: true }} />);
    wrapper.setProps({
      allPostsQuery: {
        loading: true
      }
    });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("renders in Loaded State Correctly", () => {
    const testData = {
      loading: false,
      allPosts: [
        {
          message: 'testmessage1',
          updatedAt: undefined,
          createdAt: '2018-04-07T20:34:04.000Z',
          id: 'testid1'
        },
        {
          message: 'testmessage2',
          updatedAt: '2018-04-07T20:34:04.000Z',
          createdAt: '2018-04-07T20:34:04.000Z',
          id: 'testid2'
        }
      ]
    };
    Enzyme.configure({ adapter: new Adapter() });
    const wrapper = shallow(<App allPostsQuery={testData} />);
    wrapper.setProps({
      allPostsQuery: {
        loading: true
      }
    });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
