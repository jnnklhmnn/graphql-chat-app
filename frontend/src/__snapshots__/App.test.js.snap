// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders in Loaded State Correctly 1`] = `
<CSSTransitionGroup
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={1000}
  transitionLeave={false}
  transitionName="initial-load"
>
  <div
    key="notloaded"
  />
</CSSTransitionGroup>
`;

exports[`App renders in Loading State Correctly 1`] = `
<CSSTransitionGroup
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={1000}
  transitionLeave={false}
  transitionName="initial-load"
>
  <div
    key="notloaded"
  />
</CSSTransitionGroup>
`;
